### DISCUSSION TITLE
Does this theme use cookies by default?

### DISCUSSION THREAD

**Discussion by sribooshan** [alshedivat/al-folio#281] [Q&A] [answered]
I went through the GDPR cookie consent discussion in the "Issues" section. I came across this - https://github.community/t/cookies-if-i-use-github-pages-am-i-using-cookies/130249

I am wondering if this theme al-folio uses cookies. If yes, how do I disable non-essential cookies?
I will be grateful if someone could help me out. I am very new to this website development area.

Thanks,
Booshan.

**Comment by alshedivat** [comment, marked as answer]
Unless you are using Google Analytics or the Twitter plugin, the theme does not use cookies.

You can probably check by pasting `https://alshedivat.github.io/al-folio/` in https://www.cookieserve.com/.

**Reply by sribooshan** [reply]
Thank you very much!!

**Reply by liberomureddu** [reply]
> Unless you are using Google Analytics or the Twitter plugin, the theme does not use cookies.
> 
> You can probably check by pasting `https://alshedivat.github.io/al-folio/` in https://www.cookieserve.com/.

Hi, I just discovered today that there's a local storage element, a key-value pair 'theme, system', which is related to the dark-light-system theme selection (in my case, the theme reflects the user's system theme). I think it is generated here: [theme.js](https://github.com/alshedivat/al-folio/blob/caf9976175684ff8c9df6859ea192350df1e9ce1/assets/js/theme.js).

I'm an EU resident, so I should inform the users about it, as this is not an option explicitly requested by the user, but something that happens when the user visits the site. As a solution, I don't mind providing the user with a default light theme, while keeping the possibility of switching between dark and light themes. 

Any suggestion on which areas of the code I should look at and if this would be a good solution?

If nothing works, I'll try to install a cookie information banner. Thank you!

**Reply by george-gca** [reply]
The easiest solution would be to change here to use light mode as default when no theme is already set

https://github.com/alshedivat/al-folio/blob/caf9976175684ff8c9df6859ea192350df1e9ce1/assets/js/theme.js#L213

We had a PR for this https://github.com/alshedivat/al-folio/pull/223 but it was closed. Apparently there is an easy way to implement this with https://github.com/alshedivat/al-folio/issues/199#issuecomment-2081572832.

**Reply by liberomureddu** [reply]
Thanks, that's very helpful! I just realised I'll have to go with [https://github.com/orestbida/cookieconsent](https://github.com/orestbida/cookieconsent) as suggested as I have few embedded YouTube and Bandcamp links, unless of course I remove the embeds.
